web/src/lib/components/admin-page/restore-dialoge.svelte

```diff
@@ -11,10 +12,15 @@
   }>();
 
   const handleRestoreUser = async () => {
-    const { deletedAt } = await restoreUser({ id: user.id });
-    if (deletedAt == undefined) {
-      dispatch('success');
-    } else {
+    try {
```

Comment:
Depending on how long it takes for the background job to actually clean up the user, it will most likely be shown on the ui as pending deletion (get user list happens before job runs). Noticed that the ui wasn't catching the exception and just hanging when trying to restore a user that no longer existed.

We could add some sort of force delete column to the user object to handle this edge case (filter them out from the list call).

Reply:
Maybe we can just compare the deleted at timestamp with the delete duration to know if it is pending immediate deletion or not.

Reply:
Although I guess force delete can be used independently of that... Hmm... I don't really want to add a column just for this. I would be down to send a web socket event and have the page wait up to a few seconds to receive it.

Reply:
Yeah, I didn't want to add a column either. If you have any examples of the websocket being used, I can take a look at that. But even with the websocket, the user will see the forced deleted user in the table with the restore button. Once the job is on the queue, even if they restore, the user will eventually get deleted.

Reply:
Maybe we should add something to the message that says the action is irreversible.

Reply:
Yeah, I was pointing out that the restore option could be misleading/confusing. The user can see it and click it, but at that point, the job is already in the queue and the delete is going to happen (even if they quickly click restore).

The ideal behavior is the user doesn't show up in the row after the confirmation modal is closed, but I'm not sure how we can do that without adding/setting something on the user table.

Reply:
Maybe we can change the job to still require the user be soft delete either way. If the restore happens before the job is processed then you are all good. As soon as it is processed the user disappears from the table. Yeah I will have to think about it some more. Feel free to ping me on discord if it is easier to discuss it over there.

Reply:
Calling it a night, will think more on it as well.

Although I like that idea, and actually I think that is already the behavior with the extra call out to `isReadyForDeletion`. I'll test it tomorrow, in that case would just need to add the webhook logic to update the user table once delete is complete.

Reply:
Maybe we can add a new column called `status` and it defaults to active. Then we can have a "delete in progress state" and maybe others in the future like inactive, disabled, locked, etc.